lsp/core/aws-lsp-core/src/language-service/awsLanguageService.ts

```diff
@@ -0,0 +1,12 @@
+import { Position, Range, TextDocument, TextEdit } from 'vscode-languageserver-textdocument'
+import { CompletionList, Diagnostic, FormattingOptions, Hover } from 'vscode-languageserver-types'
+
+// Where possible, this interface should follow the (VS Code) JSON Language Server's interface, which is close
+// to the core (VS Code) language server implementations.
+export interface AwsLanguageService {
```

Comment:
is this a subset of the vscode interface? would `Pick<>` be useful here?

Reply:
It is a subset of the [JSON Language Service](https://github.com/microsoft/vscode-json-languageservice/blob/main/src/jsonLanguageService.ts#L38)

I feel like the interface I linked to is the top level "json language server" interface. If it was a more generalized interface in the language server package (instead of the json service package) I'd be inclined to use that as the central type. I think we'll want to explore this interface concept further during the experiment phase though.

Reply:
Makes sense. Might be worth linking to that in the docstring, that kind of historical context can help comprehension.